UsageCollectorWorkerService/BackgroundWorkers/CollectorWorker.cs

```diff
@@ -0,0 +1,45 @@
+using UsageCollectorWorkerService.Models;
+using UsageCollectorWorkerService.Services.DataHolder;
+using UsageCollectorWorkerService.Services.DataSender;
+using UsageCollectorWorkerService.Services.SystemResourcesCollector;
+
+namespace UsageCollectorWorkerService.BackgroundWorkers;
+
+public class CollectorWorker : BackgroundService
+{
+    private readonly IDataHolderService _dataHolderService;
+    private readonly ISenderService _senderService;
+    private readonly ISysResCollectingService _sysResCollectingService;
+    private readonly int _durationOfCollectingInSeconds;
+    private readonly int _intervalBetweenCollectingInSeconds;
+
+    public CollectorWorker(
+        IDataHolderService dataHolderService,
+        ISenderService senderService,
+        ISysResCollectingService sysResCollectingService,
+        int durationOfCollectingInSeconds,
+        int intervalBetweenCollectingInSeconds)
+    {
+        _dataHolderService = dataHolderService;
+        _senderService = senderService;
+        _sysResCollectingService = sysResCollectingService;
+        _durationOfCollectingInSeconds = durationOfCollectingInSeconds;
+        _intervalBetweenCollectingInSeconds = intervalBetweenCollectingInSeconds;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        RootSysResUsageValues dataHolder = _dataHolderService.Instance;         
+        
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            dataHolder.UsageValues = await _sysResCollectingService.CollectResourcesAsync(
```

Comment:
Is the data in dataHolder ever cleared? If not, you can for example add method to DataHolder service to remove specific data (either identified by timespamp or add Id), and once threy are send by sender service, remove them.